astoria/managers/astprocd.py

```diff
@@ -331,19 +334,21 @@ async def run_process(self) -> None:
                         "robot.py",
                         stdin=asyncio.subprocess.DEVNULL,
                         stdout=asyncio.subprocess.PIPE,
-                        stderr=asyncio.subprocess.STDOUT,
+                        stderr=asyncio.subprocess.PIPE,
                         cwd=self._dir_path,
                         start_new_session=True,
                     )
                     if self._process is not None:
-                        if self._process.stdout is not None:
+                        if self._process.stdout is not None and \
+                                self._process.stderr is not None:
                             asyncio.ensure_future(
                                 self.logger(
-                                    self._process.stdout,
+                                    {LogEventSource.STDOUT: self._process.stdout,
+                                     LogEventSource.STDERR: self._process.stderr},
                                     initial_messages=log_message,
                                 ),
                             )
-                        else:
+                        if not all([self._process.stdout, self._process.stderr]):
```

Comment:
We're checking for this above, we should be able to use else?

Reply:
Yes. This was changed from an else as I had two consecutive ifs there at one point.